x-pack/plugins/security_solution/common/license/policy_config.ts

```diff
@@ -21,7 +24,7 @@ export const isEndpointPolicyValidForLicense = (
     return true; // currently, platinum allows all features
   }
 
-  const defaults = factory();
+  const defaults = policyFactoryWithoutPaidFeatures();
```

Comment:
I wonder do we really need the isEndpointPolicyValidForLicense check? It seems simpler to just strip the values if license is lower than platinum and save that.

Reply:
we have this check for the API guard if a user tries to modify a paid policy feature with a gold or below license. we don't actually want to change the policy for them, just recognize that they are doing something bad and send them an error.

Reply:
I see. I was more thinking it's only used here - https://github.com/elastic/kibana/blob/f4b8d621659a8c15a39f1d6112544de750d37a2a/x-pack/plugins/security_solution/server/endpoint/lib/policy/license_watch.ts#L115

And that might be redundant to do a check (it's pretty much a micro optimisation).

Reply:
ah, yeah. it is also used here: https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/server/endpoint/ingest_integration.ts#L193